src/Symfony/Component/Workflow/MarkingStore/MethodMarkingStore.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\MarkingStore;
+
+use Symfony\Component\Workflow\Exception\LogicException;
+use Symfony\Component\Workflow\Marking;
+
+/**
+ * MethodMarkingStore stores the marking with a subject's method.
+ *
+ * This store deals with a "single state" or "multiple state" Marking.
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class MethodMarkingStore implements MarkingStoreInterface
+{
+    private $property;
+    private $singleState;
+
+    /**
+     * @param string $property Used to determine methods to call
+     *                         The `getMarking` method will use `$subject->getProperty()`
+     *                         The `setMarking` method will use `$subject->setProperty(string|array $places, array $context = array())`
+     */
+    public function __construct(string $property = 'marking', bool $singleState = false)
+    {
+        $this->property = $property;
+        $this->singleState = $singleState;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMarking($subject)
+    {
+        $method = 'get'.ucfirst($this->property);
```

Comment:
I'd suggest removing the call to ucfirst: it's not needed + it can lead to issues with the turkish locale.
(we can keep the call in the "throw" path)
(same in other similar places)

Reply:
If I do that, the doc will be a bit weird. because the user will have to set `Marking` and not `marking`.
Or should we rely on the fact PHP is case insensitive for methods ?

Thus, the PropertyInfo component suffers of the same behavior : https://github.com/lyrixx/symfony/blob/8d277ce3e594c196594170e4fbbc37be98ba4f73/src/Symfony/Component/PropertyAccess/PropertyAccessor.php#L694-L700